mbed-client/m2mbase.h

```diff
@@ -481,34 +518,28 @@ protected : // from M2MReportObserver
 
     static bool build_path(StringBuffer<MAX_PATH_SIZE_4> &buffer, const char *s1, uint16_t i1);
 
+    char* stringdup(const char* s);
+
 private:
 
```

Comment:
Could this be just a composite member of object, ie. not allocated separately and accessed via pointer? That would remove one point of failure on construction and also reduce heap allocation overhead.